src/main/java/edu/hm/hafner/analysis/parser/CodeGeneratorParser.java

```diff
@@ -0,0 +1,77 @@
+package edu.hm.hafner.analysis.parser;
+
+import java.io.UncheckedIOException;
+import java.util.Iterator;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+
+import edu.hm.hafner.analysis.IssueBuilder;
+import edu.hm.hafner.analysis.IssueParser;
+import edu.hm.hafner.analysis.ParsingException;
+import edu.hm.hafner.analysis.ReaderFactory;
+import edu.hm.hafner.analysis.Report;
+import edu.hm.hafner.analysis.Severity;
+
+/**
+ * A parser for Simulink Code Generator tool log files.
+ *
+ * @author Eva Habeeb
+ */
+public class CodeGeneratorParser extends IssueParser {
+    private static final long serialVersionUID = -1251248150731418714L;
+    private static final Pattern WARNING_PATTERN = Pattern.compile("^(Warning:)(.*)");
+
+    @Override
+    public Report parse(final ReaderFactory reader) throws ParsingException {
+        try (Stream<String> lines = reader.readStream()) {
+            return parse(lines);
+        }
+        catch (UncheckedIOException e) {
+            throw new ParsingException(e);
+        }
+    }
+
+    private Report parse(final Stream<String> lines) {
+        try (IssueBuilder builder = new IssueBuilder()) {
+            Report report = new Report();
+            Iterator<String> lineIterator = lines.iterator();
+
+            while (lineIterator.hasNext()) {
+                String line = lineIterator.next();
+                Matcher matcher = WARNING_PATTERN.matcher(line);
+
+                if (matcher.matches()) {
+                    String category = setCategory(line);
+                    builder.setCategory(category);
+                    builder.setDescription(matcher.group(2));
```

Comment:
Normally a warning should have a `message` as it is shown everywhere. The description typically just is an additional information. Can't you use something as the text before a dot `.` as message?

Reply:
Changed it to message instead of Description